jwst/outlier_detection/outlier_detection.py

```diff
@@ -107,19 +111,30 @@ def _get_outlier_pars(self):
         for kw in list(self.outlierpars.keys()):
             self.outlierpars[kw] = ref_model['outlierpars_table.{0}'.format(kw)]
 
+    def build_suffix(self, **pars):
+        """ Class-specific method for defining the resample_suffix attribute
+            using a suffix specific to the sub-class.
+        """
+        # Parse any user-provided filename suffix for resampled products
+        self.resample_suffix = '_outlier_{}.fits'.format(pars.get('resample_suffix' , self.DEFAULT_SUFFIX))
+        if 'resample_suffix' in pars: del pars['resample_suffix']
+        log.info("Defined output product suffix as: {}".format(self.resample_suffix))
+        
     def do_detection(self):
         """Flag outlier pixels in DQ of input images
         """
+        self.build_suffix(**self.outlierpars)
+        
         pars = self.outlierpars
         save_intermediate_results = pars['save_intermediate_results']
         if pars['resample_data'] is True:
             # Start by creating resampled/mosaic images for each group of exposures
             sdriz = resample.ResampleData(self.input_models, single=True,
                 blendheaders=False, **pars)
-            sdriz.do_drizzle(**pars)
+            sdriz.do_drizzle()
             drizzled_models = sdriz.output_models
             for model in drizzled_models:
-                model.meta.filename += "_i2d.fits"
+                model.meta.filename += self.resample_suffix
```

Comment:
This assumes that the value of model.meta.filename only contains a root name, with no existing product suffix or "fits" extension. Is that a valid assumption?

Reply:
This assumption has been made based on currently available test data replicating what is believed to be produced by the pipeline.  A more general function/operation could be applied here if it was deemed an invalid assumption.